Update helper text for cli generate command

seancdavis opened the pull request:
Changes here:

- T-972: Give the user a notice when the `generate` command changes the `id`. I used `Colors` directly in the `generate.ts` file to give it some emphasis.
- T-997: Add example to partial match on `grouparoo generate --list`. I created this one on the fly as I was working. Seemed like it would help.
- T-971 / T-974: The description for the `generate` command is ... _lengthy_ now. I don't love the way I did this, but it does shove the info right in the dev's face. I played around with using a library like [columnify](https://www.npmjs.com/package/columnify), but it seemed a little heavy-handed. I figured we could keep it a little uglier like this and come back to it as we think more about workflows in the CLI.
- T-980: Real simple note on the mailchimp `id` destination based on our conversation from yesterday. As a dev coming in blind, I feel like that note would be enough to tell me what to do.
- T-975: Reworked the four source templates with bootstrapped properties. This was the biggest change I made. The way I formatted it would have helped me work through it yesterday. But it's a decent-sized change to how you had those files setup, so I won't be offended if you want to keep playing with it.

Commit:
[T-971] Add id description for cli generate
Also adds more help text if id is missing

Commit:
[T-972] Add notice if cli generate changes the input id

Commit:
[T-974] Add more description about templates to generate --help

Commit:
[T-997] Add helper example for partial template name match

Commit:
Add more detail on bootstrapping to source templates

Commit:
[T-980] Add note on using mailchimp id vs email

evantahler reviewed (approved):
I like all of these changes 👍 .

There's a minor philosophy question embedded here: Should we use `...` to represent all parent/child IDs in these templates?  It's a great sign to folks that they need to change it and put something there.  But, without an example value (ie: `appId: data_warehouse`), they might get lost?

I'm generally pro `...` if in a comment nearby we include an example.  If we want to go this route, then I would suggest changing *all* the templates to match.  There are others...

evantahler commented:
Added `enhancement` label

Commit:
More placeholder values in code config templates

seancdavis commented:
@evantahler I added a handful of placeholders to these templates. It's not exhaustive, but it moves a little closer to being consistent with usage across the board. I'm thinking it's enough to wrap these tasks and move toward v0.2, though I'd like to discuss potential steps down the road to make these even more consistent.

Probably worth another scan since I adjusted so many files with the [latest commit](https://github.com/grouparoo/grouparoo/pull/1203/commits/81e0d31d6267fc5ee8331bab115ce707faac2c13).

evantahler reviewed (approved):
Good pattern: `appId: "...", // The ID of the App this Source uses - e.g. appId: "facebookApp"`

I think you found enough of them to move this forward!
